### PR TITLE
#30 #30 #31 upgrade to Apigee and other minor updates

### LINKED ISSUE #32: Upgrade to Apigee

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #32 #32 #30 

Test File: https://docs.google.com/spreadsheets/d/1lqRqvKiGbjODZRneMdxisN7NzHAVGXriEF-n9tyLcr8/edit#gid=0

Run the Job and test separately as this for clarity

`docker compose up job`
`docker compose up test` - Total 48 tests

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
             "program": "${file}",
             "console": "integratedTerminal",
             "justMyCode": false,
+            "envFile": "${userHome}/secrets/ipe/.env"
```
Comment (pushyamig) [inline .vscode/launch.json:15]: This change was made for VScode debugging. Nothing will impact the Prod run

File: api_handler/api_calls.py [hunk 2]
```diff
@@ -37,8 +37,6 @@ def check_if_response_successful(self, response: Response) -> bool:
                 logger.error('JSONDecodeError encountered')
                 response_successful = False
 
-        if not response_successful:
-            logger.error(response.text)
```
Comment (pushyamig) [inline api_handler/api_calls.py:41]: This statement was just printing out the JSON error response text, which is just the big JSON text returned. So this statement create noise in the log, hence deleting it

File: requirements.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
-python-dotenv==0.19.2
+-e git+https://github.com/tl-its-umich-edu/api-utils-python@v3.0#egg=umich-api
+pandas==1.5.3
+python-dotenv==0.21.1
 mypy==0.910
-gspread==5.0.0
-pytest==6.2.5
\ No newline at end of file
+gspread==5.7.2
+pytest==7.2.1
```
Comment (pushyamig) [inline requirements.txt:6]: Upgrading the rest of the dependencies since it is 6 months

**Comment by pushyamig** [comment]
@zqian Can you review this? I have testing planned for next Web or Thur based on when you approve this. This way we will test in Prod. APIgee migration check list says that Prod run is a crucial step.

File: ipe_utils/df_utils.py [hunk 4]
```diff
@@ -27,7 +27,7 @@ def df_dosage_clean_up(df: pd.DataFrame) -> pd.DataFrame:
     """
     Dosage values should be  number, and in between 0 and 100.
     """
-    df_dosage_to_numeric = df.loc[df[COL_DOSAGE].apply(pd.to_numeric, errors='coerce').notnull()]
+    df_dosage_to_numeric = df.loc[df[COL_DOSAGE].apply(pd.to_numeric, errors='coerce').notnull()].copy()
```
Comment (zqian) [inline ipe_utils/df_utils.py:30]: I think the old code without .copy() should also work. But the new code works, too.

**Review by zqian — APPROVED** [review]
All look good!

File: ipe_utils/df_utils.py (hunk 4 above)
Comment (pushyamig) [reply ipe_utils/df_utils.py:30]: It will work, but it throws a warning, so this change is to remove the warning.

**Pull request merged**